### PR TITLE
SQL Extraction: highlight queries in VS Code

### PR DESCRIPTION
**Pull request by creativitRy**
Highlights all query fragments and query usages. Different queries are highlighted in different colors.
image

Also un-gitignored the necessary .vscode folder that was autogenerated.

### CODE REVIEW

File: tools/vscode_sql_extraction/.gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ node_modules
 .vscode-test/
 *.vsix
 /resources/sql_extraction
+!/.vscode/
```
Comment (feiling) [inline tools/vscode_sql_extraction/.gitignore:6]: Are all those files under the .vscode directory needed to build the plugin?

File: tools/vscode_sql_extraction/src/highlighter.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      return filePath === openPath;
+    });
+  }
+
+  /**
+   * Updates the cache if needed, and then returns the relevant decoration.
+   *
+   * @param index decoration number to return.
+   */
+  private getColor(index: number): vscode.TextEditorDecorationType {
```
Comment (feiling) [inline tools/vscode_sql_extraction/src/highlighter.ts:88]: In another pull request, please change getColor() to use the highlight color of the current theme.

File: tools/vscode_sql_extraction/.gitignore (hunk 1 above)
Comment (creativitRy) [reply tools/vscode_sql_extraction/.gitignore:6]: I believe so. The samples on GitHub all have those 4 files.

File: tools/vscode_sql_extraction/src/highlighter.ts (hunk 2 above)
Comment (creativitRy) [reply tools/vscode_sql_extraction/src/highlighter.ts:88]: Good idea! Currently, I'm highlighting each query using a different color, but I can just have a few theme colors and cycle between them.

**Pull request merged**